EIPS/eip-7620.md

```diff
@@ -0,0 +1,279 @@
+All of these alternatives either complicated the otherwise simple data structures or took away useful features (like the dynamically sized portion of the data section).
+    
+`CREATE4` has two "light" failure modes in case the initcontainer is not present and in case the EOF validation is unsuccessful. An alternative design where both cases led to a "hard" failure (consuming the entire gas available) was considered. We decided to have the more granular and forgiving failure modes in order to align the gas costs incurred to the actual work the EVM performs.
+
+EOF contract creation requires the Creator Contract be introduced via a state change, because neither legacy contracts nor create transactions can deploy EOF code. The alternative approach which was to continue using legacy creation would still rely on fetching the *initcode* from memory and not satisfy the requirement of code non-observability.
+
+## Backwards Compatibility
+
+This change poses no risk to backwards compatibility, as it is introduced at the same time EIP-3540 is. The new instructions are not introduced for legacy bytecode (code which is not EOF formatted), and the contract creation options do not change for legacy bytecode.
+    
+Legacy create transactions with `data` starting with `EF00` now are invalid. Similarly `CREATE` and `CREATE2` calls with `EF00` initcode fail early without executing the initcode. Previously, in both cases the initcode execution would begin and fail on the first undefined instruction `EF`.
+
+## Test Cases
+
+Creation transaction, `CREATE` and `CREATE2` cannot have its *code* starting with `0xEF`, but such cases are covered already in [EIP-3541](./eip-3541.md). However, new cases must be added where creation transaction, `CREATE` or `CREATE2` have its *initcode* being (validly or invalidly) EOF formatted:
+    
+| Initcode | Expected result |
+| - | - |
+| `0xEF` | initcode starts execution and fails |
+| `0xEF01` | initcode starts execution and fails |
+| `0xEF5f` | initcode starts execution and fails |
+| `0xEF00` | invalid creation transaction (or `CREATE` / `CREATE2` fails early, returns 0 and keeps sender nonce intact) |
+| `0xEF0001` | as above |
+| valid EOFv1 container | as above |
+
+Since EOF contract validation happens for all EOF containers during `CREATE4`, the following cases must be tested:
+    
+- `CREATE4` references a valid EOF `initcontainer` having only valid subcontainers
+- `CREATE4` references legacy code, contract creation fails
+- `CREATE4` references an invalid EOF `initcontainer`, contract creation fails
+- `CREATE4` references a valid EOF `initcontainer` having an invalid EOF subcontainer somewhere in the subcontainer tree, contract creation fails
+- `CREATE4` references a valid EOF `initcontainer` having legacy code somewhere in the subcontainer tree, contract creation fails
+  
+Cases for initcode calling "nested" `CREATE3` or `CREATE4` in various combinations
+    
+## Security Considerations
+
+Needs discussion.
```

Comment:
```suggestion
Needs discussion. <!-- TODO -->
```

If you use an HTML-style comment, the linter will complain when you move to review.